HTTPServer/HttpProcessor.cs

```diff
@@ -212,7 +214,15 @@ public void HandleClient(TcpClient tcpClient)
                                                     if (string.IsNullOrEmpty(res.Item1))
                                                         response = HttpBuilder.InternalServerError();
                                                     else
-                                                        response = HttpResponse.Send(res.Item1, "text/xml");
+                                                    {
+                                                        response.Headers.Add("Date", DateTime.Now.ToString("r"));
+                                                        statusCode = HttpStatusCode.OK;
+                                                    }
+                                                    response.HttpStatusCode = (Models.HttpStatusCode)statusCode;
+                                                    if(!string.IsNullOrEmpty(res.Item2))
```

Comment:
This needs to check if Item 1 is null or empty.

If there is a result we put the result content type, aka item 2, if not, use text/plain as a fallback for status 500

Reply:
Never mind, it is valid, as no content should return an empty content type anyway.